lib/host/inject.js

```diff
@@ -4,28 +4,55 @@ var path = require('path');
 var url = require('url');
 var mime = require('mime-types');
 
+function includes(str, search) {
+  return str.indexOf(search) != -1;
+}
+
+function ensureDoctype(html) {
+  const hasDoctype = includes(html, '<!doctype') || includes(html, '<!DOCTYPE');
+  if (hasDoctype) {
+    return html;
+  }
+
+  console.warn('Doctype not found; injecting <!doctype html> to prevent quirksmode problems');
+  return '<!doctype html>' + html;
+}
+
+function injectScript(html, scriptUrl) {
+  if (includes(html, scriptUrl)) {
+    console.warn('Script "' + scriptUrl + '" will be auto-injected; please do not include it yourself');
+    return html;
+  }
+
+  const script = '<script type="text/javascript" src="' + scriptUrl + '"></script>';
+
+  const hasBodyEnd = includes(html, '</body>');
```

Comment:
Not anywhere else, it looked more clear to me at the time to name it; I will write a unit test of the injector, which should be good enough/easiest.